kvserver: repro L0 snap ingestion

tbg opened the pull request:
Repro for #80589.

100% of all snaps go to L0, see log.txt.

Release note: None

nicktrav committed:
vendor: bump Pebble to b33d6e173cae

```
b33d6e17 internal/keyspan: remove bounds from FragmentIterator
07e7075b internal/keyspan: fix SeekLT bug
4c199d34 sstable: add scaffolding to enable sstable parallelism
338611b0 sstable: fix indexBlockBuf.shouldFlush by accounting for inflight data
8decd08c db: pool range-key iterator structs, remove Clone
d8ca1fc6 keyspan: implement Truncate in terms of Filter
5926e724 keyspan: add `keyspan.Filter`
7582b880 db,record: add internal metrics for throughput, latency and utilization
8b3db52b internal/keyspan: Emit empty spans where there are no range keys
4a0889f8 db: add IngestOperationStats for admission control
194770c8 internal/metamorphic/crossversion: add TestMetaCrossVersion
785bb36d internal/keyspan: clean up InterleavingIter
b9bcae65 db: generate delete compaction hints automatically in tests
4e77c077 db: update comment on L0StopWritesThreshold
99d8a549 internal/metamorphic: add support for an initial database state
e232404f vfs: add CloneSync, CloneSkip options for vfs.Clone
7accf3df internal/metamorphic: use a wal_dir within the data directory
886e91f6 internal/keyspan: tweak DefragmentReducer return types
31def7d7 db: add additional deletion only hint test case; update diagram
1793e521 sstable: enforce sstable size estimation invariants
e4927968 db: implement rangedel defragmentation using keyspan.DefragmentingIter
41895ae0 internal/keyspan: reduce over defragmented spans
c55c1d80 db: check ordering invariant for sublevel compactions
322f7e9c internal/keyspan: genercize masking
cca64783 compaction: add L0CompactionFileThreshold compaction trigger
5cd60640 db: respect changed filters in SetOptions
8cc87859 db: add additional table cache test for the Open function
c8e7d414 db: add range key count to table stats
ec4dcaa7 db: factor out some table stats helpers
fb7ccaa2 internal/arenaskl: Overflow fixes
53fd136b internal/keyspan: move and refactor range key iterators as FragmentIterators
c6ff6904 db: use sublevels for l0 manual compactions
8e8ff2be db: add Iterator.SetOptions for modifying iterator options
1b227d2c db: factor out point iterator initialization
3e86958e docs/rfc: flushable ingested sstable addition
0b5f42a0 internal/keyspan: Add LevelIter for range keys
d7fb4eb2 docs/rfc: flushable ingested sstable
92c22536 db: remove unused variable from loop
7ad3551b *: refactor keyspan.Span to represent a set of keys
```

Release note:

tbg committed:
kvserver: repro L0 snap ingestion

Repro for #80589.

100% of all snaps go to L0, see log.txt.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80611)
<!-- Reviewable:end -->

tbg committed:
Check for 100mb range

Looks like the chunky part doesn't go to L0.

```
W220428 13:55:54.088449 603 kv/kvserver/pkg/kv/kvserver/replica_raftstorage.go:952 ⋮ [n3,s3,r45/2:{-}] 65  XXX ingestion into L0: {Bytes:129007245 ApproxIngestedIntoL0Bytes:6311}
W220428 13:55:59.208826 200 kv/kvserver/pkg/kv/kvserver/replica_raftstorage.go:952 ⋮ [n2,s2,r45/3:{-}] 67  XXX ingestion into L0: {Bytes:129007393 ApproxIngestedIntoL0Bytes:6459}
```

Release note: None

tbg committed:
toolbox to run against real cluster

Release note: None
